python/tvm/target.py

```diff
@@ -428,6 +415,38 @@ def opengl(options=None):
     return _api_internal._TargetCreate("opengl", *options)
 
 
+def arm_cpu(model='unknown', options=None):
+    """Returns a ARM CPU target.
+    This function will also donwload pre-tuned op parameters
+
+    Parameters
+    ----------
+    model: str
+        SoC name or phone name of the arm board.
+    options : str or list of str
+        Additional options
+    """
+    opt_table = {
+        "pixel2":    ["-model=snapdragon835", "-target=arm64-linux-android"],
+        "mate10":    ["-model=kirin970", "-target=arm64-linux-android"],
+        "mate10pro": ["-model=kirin970", "-target=arm64-linux-android"],
+        "p20":       ["-model=kirin970", "-target=arm64-linux-android"],
+        "p20pro":    ["-model=kirin970", "-target=arm64-linux-android"],
+        "rasp3b":    ["-model=bcm2837", "-target=armv7l-linux-gnueabihf"],
+        "rk3399":    ["-model=rk3399", "-target=aarch64-linux-gnu"],
+        "pynq":      ["-model=pynq", "-target=armv7a-linux-eabi"],
+    }
+    pre_defined_opt = opt_table.get(model, [])
+
+    if not os.path.isfile(os.path.join(AUTOTVM_PRETUNED_PARAM_ROOT_PATH, "arm_cpu.log")):
```

Comment:
consolidate all the logics of file system manipulation and autotvm cache into one file, say autotvm.tophub